lib/internal/modules/esm/translators.js

```diff
@@ -291,9 +286,9 @@ translators.set('builtin', async function builtinStrategy(url) {
 });
 
 // Strategy for loading a JSON file
-translators.set('json', async function jsonStrategy(url) {
+translators.set('json', async function jsonStrategy(url, source) {
   emitExperimentalWarning('Importing JSON modules');
-  debug(`Translating JSONModule ${url}`);
+  assertBufferSource(source, false, 'load');
```

Comment:
This needs to permit strings:

```suggestion
  assertBufferSource(source, true, 'oad');
```